std-bits/table/src/main/java/org/enso/table/data/column/builder/DateTimeBuilder.java

```diff
@@ -75,11 +90,29 @@ public void appendBulkStorage(Storage<?> storage) {
 
   @Override
   public boolean accepts(Object o) {
-    return o instanceof ZonedDateTime;
+    return o instanceof ZonedDateTime || (allowDateToDateTimeConversion && o instanceof LocalDate);
   }
 
   @Override
   protected Storage<ZonedDateTime> doSeal() {
     return new DateTimeStorage(data, currentSize);
   }
+
+  @Override
+  public void retypeToMixed(Object[] items) {
```

Comment:
I wonder if we shouldn't just have a builder that makes conversion decisions at the end, rather than having to handle special cases of converting and possibly converting back.